sim/warlock/destruction/destruction.go

```diff
@@ -25,21 +25,64 @@ func RegisterDestructionWarlock() {
 
 func NewDestructionWarlock(character *core.Character, options *proto.Player) *DestructionWarlock {
 	destroOptions := options.GetDestructionWarlock().Options
-	destroLock := &DestructionWarlock{
+	destruction := &DestructionWarlock{
 		Warlock: warlock.NewWarlock(character, options, destroOptions.ClassOptions),
 	}
 
-	return destroLock
+	return destruction
 }
 
 type DestructionWarlock struct {
 	*warlock.Warlock
 }
 
-func (destroLock *DestructionWarlock) GetWarlock() *warlock.Warlock {
-	return destroLock.Warlock
+func (destruction DestructionWarlock) getMasteryBonus() float64 {
+	return 0.11 + 0.0135*destruction.GetMasteryPoints()
 }
 
-func (destroLock *DestructionWarlock) Reset(sim *core.Simulation) {
-	destroLock.Warlock.Reset(sim)
+func (destruction *DestructionWarlock) GetWarlock() *warlock.Warlock {
+	return destruction.Warlock
+}
+
+func (destruction *DestructionWarlock) Initialize() {
+	destruction.Warlock.Initialize()
+
+	destruction.registerConflagrateSpell()
+}
+
+func (destruction *DestructionWarlock) ApplyTalents() {
+	destruction.Warlock.ApplyTalents()
+
+	// Mastery: Fiery Apocalypse
+	masteryMod := destruction.AddDynamicMod(core.SpellModConfig{
+		Kind:   core.SpellMod_DamageDone_Pct,
+		School: core.SpellSchoolFire,
+	})
+
+	destruction.AddOnMasteryStatChanged(func(sim *core.Simulation, oldMastery float64, newMastery float64) {
+		masteryMod.UpdateFloatValue(destruction.getMasteryBonus())
+	})
+
+	core.MakePermanent(destruction.GetOrRegisterAura(core.Aura{
+		Label:    "Mastery: Fiery Apocalypse",
+		ActionID: core.ActionID{SpellID: 77220},
+		OnGain: func(aura *core.Aura, sim *core.Simulation) {
+			masteryMod.UpdateFloatValue(destruction.getMasteryBonus())
+			masteryMod.Activate()
+		},
+		OnExpire: func(aura *core.Aura, sim *core.Simulation) {
+			masteryMod.Deactivate()
+		},
+	}))
+
+	// Cataclysm
+	destruction.AddStaticMod(core.SpellModConfig{
+		Kind:       core.SpellMod_DamageDone_Pct,
```

Comment:
Should this be Pct and not Flat if its affecting warlock spells only?

Reply:
I'm not sure on that.